src/compiler/scanner.ts

```diff
@@ -11,6 +11,11 @@ namespace ts {
         return token >= SyntaxKind.Identifier;
     }
 
+    /* @internal */
+    export function tokenIsIdentifierOrKeywordOrGreaterThan(token: SyntaxKind): boolean {
+        return token === SyntaxKind.GreaterThanToken || token >= SyntaxKind.Identifier;

```

Comment:
Call tokenIsIdentifierOrKeyword in the right arm in case that logic changes

Reply:
Fixed.